drivers/sensor/hts221/Kconfig

```diff
@@ -5,9 +5,9 @@ menuconfig HTS221
 	bool "HTS221 temperature and humidity sensor"
 	default y
 	depends on DT_HAS_ST_HTS221_ENABLED
+	depends on HAS_STMEMSC
 	select I2C if $(dt_compat_on_bus,$(DT_COMPAT_ST_HTS221),i2c)
 	select SPI if $(dt_compat_on_bus,$(DT_COMPAT_ST_HTS221),spi)
-	select HAS_STMEMSC
 	select USE_STDC_HTS221
 	help
 	  Enable driver for HTS221 I2C/SPI-based temperature and humidity sensor.
```

Comment:
Assuming hts221 needs the hal_st repo present, here is a rough idea:
```
menuconfig HTS221
	bool "HTS221 temperature and humidity sensor"
	default y
	depends on DT_HAS_ST_HTS221_ENABLED
	depends on ZEPHYR_ST_MODULE
	select I2C if $(dt_compat_on_bus,$(DT_COMPAT_ST_HTS221),i2c)
	select SPI if $(dt_compat_on_bus,$(DT_COMPAT_ST_HTS221),spi)
	select USE_STDC_HTS221
	help
	  Enable driver for HTS221 I2C/SPI-based temperature and humidity sensor.
```

Reply:
Ok, need to refer to ```ZEPHYR_ST_MODULE```. Thx